torch_geometric/loader/temporal_dataloader.py

```diff
@@ -26,13 +29,44 @@ def __init__(self, data: TemporalData, batch_size: int = 1, **kwargs):
 
         self.data = data
         self.events_per_batch = batch_size
-
+        self.device = device
+        self.negative_sampling = negative_sampling
+        if self.negative_sampling:
```

Comment:
Can we pass a `neg_sampling_fn` here instead? I feel this assumption of `min_dst_idx` and `max_dst_idx` is not very general and very dataset specific.